pallets/api/src/messaging/mod.rs

```diff
@@ -459,8 +518,36 @@ pub mod pallet {
 			let origin = ensure_signed(origin)?;
 
 			for id in &messages {
-				let message = Messages::<T>::get(&origin, id).ok_or(Error::<T>::MessageNotFound)?;
-				message.try_remove(&origin, id).and_then(|_| message.release_deposit(&origin))?;
+				let Some(message) = Messages::<T>::get(&origin, id) else {
```

Comment:
I see a small issue with this implementation, particularly around the origin of removing these messages.

Assume that a callback is NOT registered. A response is received that a contract relies on. Its offchain application is listening for the response, and then is going to read the state to get the response.

However, before that transaction can be sent to the contract, a malicious actor calls this remove function before the contract can access the data.

Seems this data should only be removable if (a couple ideas):
- The contract is the calling origin -- seems more limiting
- The data has been "claimed" or "pulled" from storage by the contract already.

Reply:
Since origin is a key in Messages Alice cannot delete messages created by Bob (or a contract for that matter). 

Assuming the origin here is the contract, i see how a malicious contract can remove a response before polling has occured...
The claiming idea may have issues of its own but at least will allow responses to have a bit more resistance to this.

Reply:
I am going to make this into its own issue, since its not related to xcm ill push it into the next sprint.

Reply:
https://app.shortcut.com/r0gue/story/3341/allow-removal-of-messages-with-responses-to-be-removed-only-if-they-have-been-polled